WcaOnRails/app/controllers/posts_controller.rb

```diff
@@ -5,7 +5,13 @@ class PostsController < ApplicationController
   before_action -> { redirect_to_root_unless_user(:can_create_posts?) }, except: [:index, :rss, :show]
 
   def index
-    @posts = Post.where(world_readable: true).order(sticky: :desc, created_at: :desc).includes(:author).page(params[:page])
+    tag = params[:tag]
+    if tag
+      @posts = Post.joins(:post_tags).where('post_tags.tag = ?', tag)
+    else
+      @posts = Post.joins("LEFT JOIN post_tags ON post_tags.post_id = posts.id").group("posts.id").having("IFNULL(SUM(tag = 'wdc'), 0) = 0")
```

Comment:
> Sorry, I'm not sure what this is asking.

It's asking if the 'wdc' tag can only be the only tag for a post?  With this code, if a post is tagged by 'wdc' and 'foo', you could see the it searching for 'foo'. And if 'wdc' posts are not supposed to be shown, that would be bad.

But it sounds like it's not about keeping things secret but what's interesting enough to be on the home page. If so, my question has been answered, and the code does what it should.